src/gsy_e/models/market/two_sided.py

```diff
@@ -320,36 +317,22 @@ def match_recommendations(
                     # re-raise exception to be handled by the external matcher
                     raise invalid_bop_exception
 
-            market_offers = iter(market_offers)
-            market_bids = iter(market_bids)
-            market_offer = next(market_offers, None)
-            market_bid = next(market_bids, None)
-
-            while market_bid and market_offer:
-                original_bid_rate = market_bid.original_price / market_bid.energy
-                trade_bid_info = TradeBidOfferInfo(
-                    original_bid_rate=original_bid_rate,
-                    propagated_bid_rate=market_bid.energy_rate,
-                    original_offer_rate=market_offer.original_price / market_offer.energy,
-                    propagated_offer_rate=market_offer.energy_rate,
-                    trade_rate=original_bid_rate)
-
-                bid_trade, offer_trade = self.accept_bid_offer_pair(
-                    market_bid, market_offer, clearing_rate,
-                    trade_bid_info, min(selected_energy, market_offer.energy, market_bid.energy))
-                were_trades_performed = True
-                if offer_trade.residual:
-                    market_offer = offer_trade.residual
-                else:
-                    market_offer = next(market_offers, None)
-                if bid_trade.residual:
-                    market_bid = bid_trade.residual
-                else:
-                    market_bid = next(market_bids, None)
-                recommendations = (
-                    self._replace_offers_bids_with_residual_in_recommendations_list(
-                        recommendations, offer_trade, bid_trade)
-                )
+            original_bid_rate = market_bid.original_price / market_bid.energy
+            trade_bid_info = TradeBidOfferInfo(
+                original_bid_rate=original_bid_rate,
+                propagated_bid_rate=market_bid.energy_rate,
+                original_offer_rate=market_offer.original_price / market_offer.energy,
+                propagated_offer_rate=market_offer.energy_rate,
+                trade_rate=original_bid_rate)
+
+            bid_trade, offer_trade = self.accept_bid_offer_pair(
+                market_bid, market_offer, clearing_rate,
+                trade_bid_info, min(selected_energy, market_offer.energy, market_bid.energy))
```

Comment:
Minor but better to have an small variable to keep the selected_energy
```
selected_energy=min(selected_energy, market_offer.energy, market_bid.energy)
```

```suggestion
                trade_bid_info, selected_energy)
```

Reply:
I think that we should avoid creating variables for values that aren't reused, pylint will complain about this too